### PR TITLE
ORC-902: Fix hadoop-common  dependency scope in orc-example

### PR DESCRIPTION
**Pull request by guiyanakuang**
### What changes were proposed in this pull request?

run orc-example, Throw an exception
```log
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/hadoop/conf/Configuration
	at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
	at java.lang.Class.privateGetMethodRecursive(Class.java:3048)
	at java.lang.Class.getMethod0(Class.java:3018)
	at java.lang.Class.getMethod(Class.java:1784)
	at sun.launcher.LauncherHelper.validateMainClass(LauncherHelper.java:650)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:632)
Caused by: java.lang.ClassNotFoundException: org.apache.hadoop.conf.Configuration
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 7 more
```

mvn dependency:tree in orc-example

```log
...
[INFO] +- org.apache.hadoop:hadoop-common:jar:2.2.0:provided
...
```
Need to set scope to compile in orc-example/pom.xml

### Why are the changes needed?

The example of orc-example cannot be run

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Thank you for making a PR, @guiyanakuang !

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you. This is caused by https://github.com/apache/orc/pull/708

Are the other examples good?

File: java/examples/pom.xml [hunk 1]
```diff
@@ -50,6 +50,7 @@
     <dependency>
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-common</artifactId>
+      <scope>compile</scope>
```
Comment (dongjoon-hyun) [inline java/examples/pom.xml:53]: We need to fix line 57 for `hadoop-hdfs`, too. Could you fix it together?

Comment (guiyanakuang) [reply java/examples/pom.xml:53]: OK

**Comment by guiyanakuang** [comment]
None of the 4 examples can run, because they all need new Configuration().

**Comment by dongjoon-hyun** [comment]
Now, every example runs?
> None of the 4 examples can run, because they all need new Configuration().

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. I verified all three commands. Thanks!

```
$ java -jar target/orc-examples-1.8.0-SNAPSHOT-uber.jar
ORC Java Examples

usage: java -jar orc-examples-*.jar [--help] [--define X=Y] <command> <args>

Commands:
   write - write a sample ORC file
   read - read a sample ORC file
   write2 - write a sample ORC file with a map
```

**Comment by dongjoon-hyun** [comment]
Merged to main/1.7.

**Pull request merged**